### ISSUE TITLE
`text-2.1.2` broke `parsers-0.12.11`

### ISSUE THREAD

**Issue by geekosaur** [ekmett/parsers#90]
There is a new export, which you are now getting because `import … hiding` doesn't exclude them.
```
[7 of 8] Compiling Text.Parser.Token ( src/Text/Parser/Token.hs, dist/build/Text/Parser/Token.o )

src/Text/Parser/Token.hs:571:52: error:
Error:     Ambiguous occurrence ‘show’
    It could refer to
       either ‘Prelude.show’,
              imported from ‘Prelude’ at src/Text/Parser/Token.hs:31:8-24
              (and originally defined in ‘GHC.Show’)
           or ‘Data.Text.show’,
              imported from ‘Data.Text’ at src/Text/Parser/Token.hs:114:1-81
    |
571 |    notFollowedBy (_styleLetter s) <?> "end of " ++ show name
    |                                                    ^^^^

src/Text/Parser/Token.hs:578:52: error:
Error:     Ambiguous occurrence ‘show’
    It could refer to
       either ‘Prelude.show’,
              imported from ‘Prelude’ at src/Text/Parser/Token.hs:31:8-24
              (and originally defined in ‘GHC.Show’)
           or ‘Data.Text.show’,
              imported from ‘Data.Text’ at src/Text/Parser/Token.hs:114:1-81
    |
578 |    notFollowedBy (_styleLetter s) <?> "end of " ++ show name
    |                                                    ^^^^

src/Text/Parser/Token.hs:586:102: error:
Error:     Ambiguous occurrence ‘show’
    It could refer to
       either ‘Prelude.show’,
              imported from ‘Prelude’ at src/Text/Parser/Token.hs:31:8-24
              (and originally defined in ‘GHC.Show’)
           or ‘Data.Text.show’,
              imported from ‘Data.Text’ at src/Text/Parser/Token.hs:114:1-81
    |
586 |   when (HashSet.member name (_styleReserved s)) $ unexpected $ "reserved " ++ _styleName s ++ " " ++ show name
    |                                                                                                      ^^^^
```
(https://github.com/haskell/cabal/actions/runs/11533905253/job/32107358221?pr=10477, https://github.com/haskell/cabal/actions/runs/11534133925/job/32107825720?pr=10478)

**Comment by geekosaur** [comment]
https://github.com/haskell/text/issues/623#issuecomment-2439700535 ff.

**Comment by RyanGlScott** [comment]
It turns out that I fixed this in `master` in 6ed66be7e3f14d927257b6f1a2439ec6fb9a4483, but that hasn't yet made it to a Hackage release. I'll work on making one.

**Comment by RyanGlScott** [comment]
I've uploaded [`parsers-0.12.12`](https://hackage.haskell.org/package/parsers-0.12.12) to Hackage with a fix.
